source/common/upstream/upstream_impl.cc

```diff
@@ -288,8 +310,9 @@ void ClusterImplBase::setHealthChecker(const HealthCheckerSharedPtr& health_chec
   health_checker_->start();
   health_checker_->addHostCheckCompleteCb([this](HostSharedPtr, bool changed_state) -> void {
     // If we get a health check completion that resulted in a state change, signal to
-    // update the host sets on all threads.
-    if (!block_hc_updates_ && changed_state) {
+    // update the host sets on all threads. See setInitialized() for why we block the reload
+    // on whether we are initializing or not.
+    if (initialize_callback_ == nullptr && changed_state) {
```

Comment:
Any reason to not put the initialize_callback_ check in reloadHealthyHosts to make sure it's applied consistently?  That'd make a good place for the "blocking reload during initialization" comment.

Reply:
sure